### PR TITLE
[CFM_219] Setup github actions in frontend

### PR DESCRIPTION
**Pull request by seanred360**
*************MERGED WITH CFM 126*********************************

This workflow: 

installs node deps and caches node_modules which speeds up the other jobs
- runs yarn build
- runs yarn test
- runs yarn lint

see this working in my fork here: https://github.com/seanred360/CFM_Frontend/actions/runs/4537249173

github-actions

### CODE REVIEW

**Comment by seanred360** [comment]
The tests fail because Jest has not been merged in CFM_126

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    branches: [dev]
+
+jobs:
+  Install_Dependencies:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "19"
```
Comment (doomjar) [inline .github/workflows/main.yml:17]: Should this line be replaced with `node-version-file: ".nvmrc"`?

File: .nvmrc [hunk 2]
```diff
@@ -0,0 +1 @@
+v19.8.1
```
Comment (doomjar) [inline .nvmrc:1]: We probably don't need to include the minor and patch versions, just the major version should do.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+      - name: Cache Yarn Dependencies
+        uses: actions/cache@v3
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+
+      - name: Run Lint
+        run: |
+          yarn lint
```
Comment (doomjar) [inline .github/workflows/main.yml:95]: Please add a single newline to the end of this and the .nvmrc files.

**Review by jaronaearle — CHANGES_REQUESTED** [review]
Since the success of this workflow is dependent on CFM_126, I'd suggest merging CFM_126 into this branch in order for it to be complete.

**Pull request closed without merging**